static/app/views/dashboards/manage/index.tsx

```diff
@@ -241,6 +241,21 @@ class ManageDashboards extends AsyncView<Props, State> {
     );
     this.onDashboardsChange();
     addSuccessMessage(`${dashboard.title} dashboard template successfully added.`);
+    setTimeout(() => {
+      this.loadDashboard(dashboard.title);
+    }, 500);
+  }
+
+  loadDashboard(dashboardTitle: string) {
+    const {organization, location} = this.props;
+    const {dashboards} = this.state;
+    const dashboard = dashboards?.find(val => val.title === dashboardTitle);
```

Comment:
Instead of passing down the title and using `.find()`, line 233 actually returns the dashboard data in the response so we can directly plug that into the URL you're creating below. I would recommend storing that in a variable, maybe `newDashboard` and passing down `newDashboard.id`.

Doing a `.find()` actually has trouble loading the correct dashboard when there are duplicates. In the request we send the title with what's in the template (e.g. "General Template"), but if our backend detects there's another dashboard with the same name we append "copy N" (e.g. "General Template copy 1") and `.find()` finds the wrong dashboard here

Reply:
@narsaynorath I;ve fixed the issues you mentioned! Thank you for the feedback!